Core/include/Acts/Seeding/GbtsDataStorage.hpp

```diff
@@ -9,279 +9,123 @@
-      m_etaBins.emplace_back(GbtsEtaBin<space_point_t>());
-    }
-  }
+  explicit GbtsDataStorage(const GbtsGeometry& g);
+  ~GbtsDataStorage();
 
-  int addSpacePoint(const GbtsSP<space_point_t> &sp, bool useClusterWidth) {
-    const GbtsLayer<space_point_t> *pL =
-        m_geo.getGbtsLayerByKey(sp.combined_ID);
+  int loadPixelGraphNodes(short layerIndex, const std::vector<GbtsNode>& coll,
+                          bool useML);
+  int loadStripGraphNodes(short layerIndex, const std::vector<GbtsNode>& coll);
 
-    if (pL == nullptr) {
-      return -1;
-    }
+  unsigned int numberOfNodes() const;
+  void sortByPhi();
+  void initializeNodes(bool useML);
+  void generatePhiIndexing(float dphi);
 
-    int binIndex = pL->getEtaBin(sp.SP->z(), sp.r());
-
-    if (binIndex == -1) {
-      return -2;
-    }
-
-    bool isBarrel = (pL->m_layer.m_type == 0);
-
-    if (isBarrel) {
-      float min_tau = -100.0;
-      float max_tau = 100.0;
-      // can't do this bit yet as dont have cluster width
-      if (useClusterWidth) {
-        float cluster_width = sp.ClusterWidth();
-        min_tau = 6.7 * (cluster_width - 0.2);
-        max_tau =
-            1.6 + 0.15 / (cluster_width + 0.2) + 6.1 * (cluster_width - 0.2);
-      }
-
-      m_etaBins.at(binIndex).m_vn.push_back(
-          std::make_unique<GbtsNode<space_point_t>>(
-              sp, min_tau, max_tau));  // adding ftf member to nodes
-    } else {
-      if (useClusterWidth) {
-        float cluster_width = sp.ClusterWidth();
-        if (cluster_width > 0.2) {
-          return -3;
-        }
-      }
-      m_etaBins.at(binIndex).m_vn.push_back(
-          std::make_unique<GbtsNode<space_point_t>>(sp));
-    }
-
-    return 0;
-  }
-
-  // for safety to prevent passing as copy
-  GbtsDataStorage(const GbtsDataStorage &) = delete;
-  GbtsDataStorage &operator=(const GbtsDataStorage &) = delete;
-
-  unsigned int numberOfNodes() const {
-    unsigned int n = 0;
-
-    for (auto &b : m_etaBins) {
-      n += b.m_vn.size();
-    }
-    return n;
-  }
-
-  void getConnectingNodes(std::vector<const GbtsNode<space_point_t> *> &vn) {
-    vn.clear();
-    vn.reserve(numberOfNodes());
-    for (const auto &b : m_etaBins) {
-      for (const auto &n : b.m_vn) {
-        if (n->m_in.empty()) {
-          continue;
-        }
-        if (n->m_out.empty()) {
-          continue;
-        }
-        vn.push_back(n.get());
-      }
-    }
-  }
-
-  void sortByPhi() {
-    for (auto &b : m_etaBins) {
-      b.sortByPhi();
-    }
-  }
-
-  void generatePhiIndexing(float dphi) {
-    for (auto &b : m_etaBins) {
-      b.generatePhiIndexing(dphi);
-    }
-  }
-
-  const GbtsEtaBin<space_point_t> &getEtaBin(int idx) const {
+  GbtsEtaBin& getEtaBin(int idx) {
     if (idx >= static_cast<int>(m_etaBins.size())) {
       idx = idx - 1;
     }
     return m_etaBins.at(idx);
   }
 
  protected:
-  const GbtsGeometry<space_point_t> &m_geo;
+  const GbtsGeometry& m_geo;
```

Comment:
```suggestion
  const GbtsGeometry* m_geo{};
```

generally pointers behave "nicer" as member than references. the type becomes automatically copyable and movable